langspec/xproc30/xproc.xml

```diff
@@ -4932,10 +4932,24 @@ as defined below, are removed:</para>
               prefix has been preserved, but it would be equally correct if a different
               prefix had been used.</para>
 
-<para>If the <tag class="attribute">expand-text</tag> attribute is
-not false, then each text node within the <tag>p:inline</tag> is evaluated
-as a <glossterm>text value template</glossterm>. The context node for
-these expressions is undefined.</para>
+<para>If the <tag class="attribute">expand-text</tag> attribute is not
+false, then each attribute value and text node within the
+<tag>p:inline</tag> is evaluated as a <glossterm>text value
+template</glossterm>. The context node for these expressions is
+undefined.</para>
+
+<para>Within the descendants of <tag>p:inline</tag> the attribute
+<tag class="attribute">p:expand-text</tag> (i.e., the attribute <code>expand-text</code>
+<emphasis>in the XProc namespace</emphasis>), may appear on an element.
+If it does, the specified value is used as the expand text value
+for the descendants of the element on which it occurs. The
+<tag class="attribute"> will not appear in the inline content.</tag>
+</para>
+
+<para>(A <tag class="attribute">p:expand-text</tag> attribute on an element in the
+XProc namespace is an error. If it occurs on other elements that are not
+descendants of <tag>p:inline</tag>, the attribute has no effect.)</para>
+
```

Comment:
Suggestion:
If it occurs below another <tag>p:inline</tag> within the topmost <tag>p:inline</tag>, that is, within an inlined pipeline, it will not immediately be used as the expand-text value of the element on which it occurs and it will appear on the inline content.
(It’s better to spell out known fringe cases in adavance. We could alternatively note that people should avoid `<tag class="attribute">p:expand-text</tag>` within inlined pipelines because what happens then is implementation-dependent.)

Reply:
@gimsieke I get (and support) the idea of your suggestion (treating p:inline as a descendant of a p:inline differently), but I can not see how this is reflected by the suggested prose in the specs. Could you please elaborate your idea.

Since the problem with @expand-text we dealing with here, only arises when one wants to construct an XProc pipeline, may be we should mark this and not bother more typical use cases. 
Could we solve this problem in a more transparent way if we abandon the idea of @expand-text having a boolean value? When we say it has three values we could cover the special case with @expand-text="pipeline-construction" (for something like that) to explicitly mark the special case.

Reply:
I don’t like 'true', 'false' and something else because it looks like a boolean but it isn’t. There’s no precedent to this in XProc AFAIK, therefore it may come as a surprise to authors.
Apart from this, let me try to phrase it differently.
“If `p:inline` contains an XProc pipeline that in turn contains another `p:inline`, any `@p:expand-text` within the inner `p:inline` is treated like an ordinary attribute; that is, neither will its value influence text expansion nor will it be removed from the inlined XProc pipeline.”

Reply:
Fixed.